src/ViewModel/AnnotationTeaser.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace eLife\Patterns\ViewModel;
+
+use Assert\Assertion;
+use eLife\Patterns\ArrayAccessFromProperties;
+use eLife\Patterns\ArrayFromProperties;
+use eLife\Patterns\SimplifyAssets;
+use eLife\Patterns\ViewModel;
+use Traversable;
+
+final class AnnotationTeaser implements ViewModel
+{
+    use ArrayAccessFromProperties;
+    use ArrayFromProperties;
+    use SimplifyAssets;
+
+    const RESTRICTED_ACCESS = true;
+    const RESTRICTED_ACCESS_TEXT = 'Only me';
+
+    private $content;
+    private $document;
+    private $highlight;
+    private $isReply;
+    private $meta;
+    private $inContextUri;
+
+    private function __construct(
+        string $document,
+        Date $date,
+        string $inContextUri,
+        string $highlight = null,
+        string $content = null,
+        bool $isRestrictedAccess = false,
+        bool $isReply = false
+    ) {
+        Assertion::notEmpty($document);
+        Assertion::notNull($date);
+        Assertion::notEmpty($inContextUri);
+
+        $this->document = $document;
+        $this->inContextUri = $inContextUri;
+
+        if ($isReply) {
+            $this->isReply = $isReply;
+        }
+
+        if ($highlight) {
+            $this->highlight = $highlight;
+        }
+
+        if ($content) {
+            $this->content = $content;
+        }
+
+        if ($isRestrictedAccess) {
+            $this->meta = Meta::withText(self::RESTRICTED_ACCESS_TEXT, $date);
+        } else {
+            $this->meta = Meta::withDate($date);
+        }
+    }
+
+    public static function full(
```

Comment:
Is there a better name for this? All the things probably shouldn't be optional too?

Reply:
Looking again, `$inContextUri`, `$highlight` and `$content` can't be empty.